spec/views/posts/index_spec.rb

```diff
@@ -0,0 +1,41 @@
+require 'rails_helper'
+
+RSpec.describe 'Post index page', type: :feature do
+  before(:example) do
+    @user = User.create(name: 'Elon Musk', photo: 'link/goes/here', bio: 'this is my bio')
+    @post_one = Post.create(title: 'My first post', text: 'Body text of the post', author_id: @user.id)
+    Post.create(title: 'My second post', text: 'Body text of the post', author_id: @user.id)
+    Post.create(title: 'My third post', text: 'Body text of the post', author_id: @user.id)
+    Post.create(title: 'My fourth post', text: 'Body text of the post', author_id: @user.id)
+    Comment.create(text: 'My first awesome comment straight from Kenya', author_id: @user.id, post_id: @post_one.id)
+    visit user_posts_path(@user)
+  end
+
+  it 'renders photo of the user' do
+    expect(page.html).to include(@user.photo)
+  end
+  it 'show username of a given user' do
+    expect(page).to have_content('Elon Musk')
+  end
+
+  it 'shows first comments of the post' do
+    expect(page).to have_content('My first awesome comment straight from Kenya')
+  end
+
+  it 'shows how many comments a post has' do
+    expect(page).to have_content(1)
+  end
+
```

Comment:
- [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=see%20the%20user%27s%20username.-,I%20can%20see%20the%20number%20of%20posts%20the%20user%20has%20written.,-I%20can%20see%20a), Please add a Test that asserts you can see the number of posts the user has written.
- [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=I%20can%20see%20a%20post%27s%20title.), Please add a Test that asserts you can see a post's title.
- [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=I%20can%20see%20some%20of%20the%20post%27s%20body.), Please add a Test that asserts you can see some of the post's body.

Reply:
Hi @Omar-Muhamad, thank you for taking the time to leave this excellent review. The changes above have been applied. Cheers